data/secscan_model/secscan_v4_model.py

```diff
@@ -0,0 +1,163 @@
+import logging
+import itertools
+
+from collections import namedtuple
+from math import log10
+from peewee import fn, JOIN
+
+from data.secscan_model.interface import SecurityScannerInterface
+from data.secscan_model.datatypes import ScanLookupStatus, SecurityInformationLookupResult
+from data.registry_model.datatypes import Manifest as ManifestDataType
+from data.registry_model import registry_model
+from util.migrate.allocator import yield_random_entries
+from util.secscan.api import SecurityConfigValidator
+from util.config import URLSchemeAndHostname
+
+from data.database import (
+    Manifest,
+    ManifestSecurityStatus,
+    IndexerVersion,
+    IndexStatus,
+    Repository,
+    User,
+)
+
+
+logger = logging.getLogger(__name__)
+
+# TODO(alecmerdler): Retrieve `/state` from security scanner API
+indexer_state = ""
+
+
+class ScanToken(namedtuple("NextScanToken", ["min_id"])):
+    """
+    ScanToken represents an opaque token that can be passed between runs of the security worker
+    to continue scanning whereever the previous run left off. Note that the data of the token is
+    *opaque* to the security worker, and the security worker should *not* pull any data out or modify
+    the token in any way.
+    """
+
+
+class V4SecurityScanner(SecurityScannerInterface):
+    """
+    Implementation of the security scanner interface for Clair V4 API-compatible implementations.
+    """
+
+    def __init__(self, app, instance_keys, storage):
+        self.app = app
+
+        validator = SecurityConfigValidator(
+            app.config.get("FEATURE_SECURITY_SCANNER", False),
+            app.config.get("SECURITY_SCANNER_ENDPOINT"),
```

Comment:
This endpoint field is going to have to be different for V4, and we're going to need a different validator now that i think of it.

So: Please create a new validator in this package, and have it take in a V4 endpoint, and (for now), have it always return False for `valid`

Reply:
Done